src/main/resources/nwoolcan/view/stock_detail.fxml

```diff
@@ -2,21 +2,67 @@
 
 <?import nwoolcan.view.subview.SubView?>
 <?import javafx.scene.control.Label?>
-<?import javafx.scene.layout.GridPane?>
-<?import javafx.scene.control.Button?>
 <?import javafx.scene.layout.FlowPane?>
+<?import javafx.scene.layout.BorderPane?>
+
+<?import javafx.scene.text.Font?>
+<?import javafx.scene.layout.VBox?>
+<?import javafx.scene.control.TitledPane?>
+<?import javafx.scene.layout.HBox?>
+
+<?import javafx.scene.control.Button?>
+<?import javafx.scene.control.TableView?>
 <SubView xmlns="http://javafx.com/javafx"
          xmlns:fx="http://javafx.com/fxml"
-         fx:controller="nwoolcan.view.StockDetailController"
-         fx:id="subView">
-    <FlowPane>
-        <Button text="Back" onAction="#btnBackClick"/>
-        <GridPane>
-            <Label text="Name: " GridPane.columnIndex="0" GridPane.rowIndex="0"/>
-            <Label fx:id="lblName" GridPane.columnIndex="1" GridPane.rowIndex="0"/>
-            <Label text="Id: " GridPane.columnIndex="0" GridPane.rowIndex="1"/>
-            <Label fx:id="lblId" GridPane.columnIndex="1" GridPane.rowIndex="1"/>
-        </GridPane>
-    </FlowPane>
+         fx:controller="nwoolcan.view.warehouse.stock.StockDetailController"
+         fx:id="stockDetailSubView">
+    <BorderPane>
+        <top>
+            <Label text="Stock Detail" BorderPane.alignment="CENTER">
+                <font>
+                    <Font size="28.0" />
+                </font>
+            </Label>
+        </top>
+        <center>
+            <VBox>
+                <Button text="Back" onAction="#backButtonClick"/>
+                <TitledPane text="General informations">
+                    <HBox>
+                        <VBox>
+                            <FlowPane>
```

Comment:
recommended not to use flow panes but use a vbox or a hbox since flow pane tend to wrap themselves based on the width. if what you want is to wrap tho they are what you need